### PR TITLE
Implement production dispute system

### PR DESCRIPTION
**Pull request by Chralt98**
### What does it do?

It implements ZIP-0 Version 2, Court and has a lot of fixes for Global Disputes.

### Are there relevant PRs or issues?

- https://github.com/zeitgeistpm/zeitgeist/pull/938
- https://github.com/zeitgeistpm/zeitgeist/pull/912
- https://github.com/zeitgeistpm/zeitgeist/pull/976

### CODE REVIEW

File: runtime/zeitgeist/src/lib.rs [hunk 1]
```diff
@@ -124,8 +124,7 @@ impl Contains<RuntimeCall> for IsCallable {
         use pallet_vesting::Call::force_vested_transfer;
 
         use zeitgeist_primitives::types::{
-            MarketDisputeMechanism::{Court, SimpleDisputes},
-            ScoringRule::RikiddoSigmoidFeeMarketEma,
+            MarketDisputeMechanism::SimpleDisputes, ScoringRule::RikiddoSigmoidFeeMarketEma,
```
Comment (Chralt98) [inline runtime/zeitgeist/src/lib.rs:127]:
```suggestion
            MarketDisputeMechanism::{Court, SimpleDisputes},
            ScoringRule::RikiddoSigmoidFeeMarketEma,
```

File: runtime/zeitgeist/src/lib.rs [hunk 2]
```diff
@@ -105,7 +105,7 @@ pub type ContractsCallfilter = Nothing;
 #[derive(scale_info::TypeInfo)]
 pub struct IsCallable;
 
-// Currently disables Court, Rikiddo and creation of markets using Court or SimpleDisputes
+// Currently disables Rikiddo and creation of markets using SimpleDisputes
```
Comment (Chralt98) [inline runtime/zeitgeist/src/lib.rs:108]:
```suggestion
// Currently disables Court, Rikiddo and creation of markets using Court or SimpleDisputes
```

**Review by sea212 — APPROVED** [review]
Let's go :star_struck: 

*Please also add all the relevant documents (ZIPs etc.) to the references section in this PR*

**Review by sea212 — APPROVED** [review]

**Comment by sea212** [comment]
I ran all skipped / failed checks locally (they failed due to an out of disk space error), every check passed locally.

**Pull request merged**